SC_redesign/Crop_and_Soil/soil/evapotranspiration.py

```diff
@@ -0,0 +1,264 @@
+            snow_water_content: amount of water from snow in mm
+            potential_evapotranspiration_adjusted: potential evapotranspiration adjusted for evaporation of free water
+                in canopy in mm
+            transpiration: transpiration in mm for a given day
+        Returns:
+            maximum soil evaporation in the day adjusted for plant water use
+
+        SWAT Reference: 2:2.3.7, 9
+        """
+        soil_cover_index = Evapotranspiration._determine_soil_cover_index(above_ground_biomass, residue,
+                                                                          snow_water_content)
+        max_soil_evaporation = potential_evapotranspiration_adjusted * soil_cover_index  # 2:2.3.7
+        actual_max_soil_evaporation = min(max_soil_evaporation,  # 2:2.3.9
+                                          ((max_soil_evaporation * potential_evapotranspiration_adjusted) /
+                                           (max_soil_evaporation + transpiration))
+                                          )
+        return actual_max_soil_evaporation
+
+    @staticmethod
+    def _determine_soil_cover_index(above_ground_biomass: float, residue: float, snow_water_content: float) -> float:
+        """
+        Calculate soil cover index
+
+        Args:
+            above_ground_biomass: mass of plant above ground in kg per hectare
+            residue: biomass separated from plant on the ground in kg per hectare
+            snow_water_content: amount of water from snow in mm
+
+        Returns:
+            soil cover index (unitless)
+
+        SWAT Reference: 2:2.3.8
+        """
+        if snow_water_content > 0.5:
+            return 0.5
+        else:
+            return exp((-5.0 * (10 ** (-5))) * (above_ground_biomass + residue))
+
+    @staticmethod
+    def _determine_maximum_soil_evaporation(soil_evaporation_adj: float, snow_water_content: float) -> float:
+        """Calculates the maximum amount of evaporation from soil in a given day
+
+        Args:
+            soil_evaporation_adj: maximum soil evaporation adjusted for plant water use on a given day in mm
+            snow_water_content: amount of water in the snow pack on a given day prior to accounting for sublimation, in
+            mm TODO: verify that "amount of water in the snow pack on a given day" (2:2.3.3.1) and "snow water content"
+                (2:2.3.3) mean the same thing
+
+        Returns:
+            maximum soil water evaporation on a given day in mm
+
+        SWAT Reference: 2:2.3.3.1
+        """
+        if soil_evaporation_adj < snow_water_content:
+            return 0  # 2:2.3.10
+        else:
+            return soil_evaporation_adj - snow_water_content  # 2:2.3.15
+
+    @staticmethod
+    def _determine_evaporative_demand(max_soil_water_evaporation: float, layer_data: LayerData) -> float:
+        """calculates the evaporative demand for a given layer of soil
```

Comment:
Let's indicate in the name that this is a layer function. Also, how do you feel about removing the dependency on `LayerData`? My preference is to always do so (in `@staticmethod`s) unless it would require an unreasonable number of parameters

```suggestion
    def _determine_layer_evaporative_demand(max_soil_water_evaporation: float, top_depth: float, bottom_depth: float, compensation: float) -> float:
        """calculates the evaporative demand for a given layer of soil
```

note that the acronym "esco" should be replaced with a better name. As a parameter, I'm good with simply `compensation` but the corresponding `LayerData` attribute should be something like `evapotranspiration_compensation_coefficient`

Reply:
Removing the dependency is no problem here, I'll take care of it